systemtests/run_test.py

```diff
@@ -0,0 +1,125 @@
+        bagname = "multitraj_" + bagname
+    else:
+        print("run_test.py : test not defined")
+        return
+    
+    src = "source " + bagfolder[:index-9] + "install/setup.bash" # -> "source /home/github/actions-runner/_work/crazyswarm2/crazyswarm2/ros2_ws/install/setup.bash"
+    
+    command = f"{src} && ros2 bag record -s mcap -o {bagname} /tf"
+    record_bag =  Popen(command, shell=True, cwd=bagfolder, stderr=PIPE, stdout=True, 
+                        start_new_session=True, text=True, executable="/bin/bash") 
+
+    command = f"{src} && ros2 run crazyflie_examples figure8"
+    start_flight_test = Popen(command, shell=True, stderr=True, stdout=True, 
+                              start_new_session=True, text=True, executable="/bin/bash")
+    
+
+    Waiter(testduration)  #wait x seconds for the crazyflie to fly the test
+
+    os.killpg(os.getpgid(start_flight_test.pid), signal.SIGTERM)  #kill flight test and all of its child processes
+    os.killpg(os.getpgid(record_bag.pid), signal.SIGTERM) #kill rosbag 
+
+    #if something went wrong with the bash command lines in Popen, print the error
+    if record_bag.stderr != None:
+        print(testname," record_bag stderr: ", record_bag.stderr.readlines())
+    if start_flight_test.stderr != None:
+        print(testname," start_flight flight stderr: ", start_flight_test.stderr.readlines())
+
+
+
+def translate_and_plot(testname:str, bagfolder:str):
+    '''Helper function that translates rosbag .mcap format to .csv, then uses that csv to plot a pdf '''
+    index = bagfolder.find("bag_")
+    bagname = bagfolder[index:]
+    # NB : the mcap filename is almost the same as the folder name but has _0 at the end
+    inputbag = str(bagfolder) + f"/{testname}_{bagname}_0.mcap"
+    output_csv = str(bagfolder) + f"/{testname}_{bagname}_0.csv"
+    writer = McapHandler()
+    writer.write_mcap_to_csv(inputbag, output_csv)  #translate bag from mcap to csv
+    output_pdf = str(path.parents[3].joinpath(f"results/Results_{testname}_"+ now +".pdf"))
+    rosbag_csv = output_csv
+    if "figure8" in testname:
+        test_file = "../crazyflie_examples/crazyflie_examples/data/figure8.csv"
+    elif "multi_trajectory" in testname:
+        test_file = "../crazyflie_examples/crazyflie_examples/data/multi_trajectory/traj0.csv"
+    else:
+        print("run_test.py : test file not defined")
+    plotter = Plotter()
+    plotter.create_figures(test_file, rosbag_csv, output_pdf) #plot the data
+
+
+if __name__ == "__main__":
+
+    path = Path(__file__)           #Path(__file__) should be "/home/github/actions-runner/_work/crazyswarm2/crazyswarm2/ros2_ws/src/crazyswarm2/systemtests/newsub.py" ; path.parents[0]=.../systemstests
+    
+    #delete results, logs and bags of previous experiments
+    shutil.rmtree(path.parents[3].joinpath("bagfiles"))
+    shutil.rmtree(path.parents[3].joinpath("results"))
+    shutil.rmtree(path.parents[8].joinpath(".ros/log"))
```

Comment:
This shouldn't depend on where the repo is checked out in. Use "~/.ros/log" instead.